src/rpc/blockchain.cpp

```diff
@@ -181,7 +181,8 @@ UniValue blockToJSON(BlockManager& blockman, const CBlock& block, const CBlockIn
         case TxVerbosity::SHOW_DETAILS:
         case TxVerbosity::SHOW_DETAILS_AND_PREVOUT:
             CBlockUndo blockUndo;
-            const bool have_undo{WITH_LOCK(::cs_main, return !blockman.IsBlockPruned(blockindex) && UndoReadFromDisk(blockUndo, blockindex))};
+            const bool is_not_pruned{WITH_LOCK(::cs_main, return !blockman.IsBlockPruned(blockindex))};
+            const bool have_undo{is_not_pruned && UndoReadFromDisk(blockUndo, blockindex)};
```

Comment:
efd82aec8a2dd0fca8f2597c3f84cefe057d1243

nit: Less negations:
```suggestion
            const bool is_pruned{WITH_LOCK(::cs_main, return blockman.IsBlockPruned(blockindex))};
            const bool have_undo{!is_pruned && UndoReadFromDisk(blockUndo, blockindex)};
```
?

Reply:
How is this less negations? You're swapping a negation on one line, for a negation on a different line.

Reply:
> How is this less negations?

line 184: `is_not_pruned`
line 184: `!blockman.IsBlockPruned(blockindex)`
line 185: `is_not_pruned `
total = 3 cases

--

in suggestion
line 185: `!is_pruned`
total =1 case

Reply:
I wouldn't consider variable names describing a state "negations", and I'm not sure why there is a need to avoid them.

Reply:
Can also keep everything in one line as-is, and reduce the diff to a one-char diff.


```cpp
const bool have_undo{WITH_LOCK(::cs_main, return !blockman.IsBlockPruned(blockindex)) && UndoReadFromDisk(blockUndo, blockindex)};

Reply:
nm, I agree with the current code.